database/sqlite/sqlite_context.c

```diff
@@ -0,0 +1,597 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#include "sqlite_functions.h"
+#include "sqlite_context.h"
+#include "sqlite_db_migration.h"
+
+#define DB_CONTEXT_METADATA_VERSION 2
+
+const char *database_context_config[] = {
+    "CREATE TABLE IF NOT EXISTS context (host_id BLOB, id TEXT, version INT, title TEXT, chart_type TEXT, " \
+    "unit TEXT, priority INT, first_time_t INT, last_time_t INT, deleted INT, family TEXT, PRIMARY KEY (host_id, id));",
+
+    NULL
+};
+
+const char *database_context_cleanup[] = {
+
+    NULL
+};
+
+sqlite3 *db_context_meta = NULL;
+
+/*
+ * Initialize the SQLite database
+ * Return 0 on success
+ */
+int sql_init_context_database(int memory)
+{
+    char sqlite_database[FILENAME_MAX + 1];
+    int rc;
+
+    if (likely(!memory))
+        snprintfz(sqlite_database, FILENAME_MAX, "%s/context-meta.db", netdata_configured_cache_dir);
+    else
+        strcpy(sqlite_database, ":memory:");
```

Comment:
Why are we using `strcpy` instead [strncpyz](https://github.com/netdata/netdata/blob/ba52f5d91f61389718e57502a94078b5beaae34f/libnetdata/inlined.h#L154-L163). 

I also suggest to use `:memory:` as a preprocessor.

If you do not want to use `strncpyz`, I suggest to use `memcpy`, because we cannot garantee that all available versions of `strcpy` will also copy the `\0` character.